cdap-ui/app/hydrator/home-ctrl.js

```diff
@@ -45,22 +45,42 @@ angular.module(PKG.name + '.feature.hydrator')
         .then(function (latest) {
 
           if (latest && checkNamespace(latest)) {
-            $state.go('hydrator.list', {namespace: latest}, {reload: true});
+            $window.location.href = $window.getHydratorUrl({
```

Comment:
Looks like namespace in stateParams is the only delta in the if-else conditions here. You can use a local variable to set namespace and do `$window.location.href = ....`  only once so that it's easier to read?